cni.go

```diff
@@ -154,16 +154,39 @@ func (c *libcni) Setup(ctx context.Context, id string, path string, opts ...Name
 	return c.createResult(result)
 }
 
+type asynchAttachResult struct {
+	res *types100.Result
+	err error
+}
+
+func asynchAttach(ctx context.Context, n *Network, ns *Namespace, wg *sync.WaitGroup, rc chan asynchAttachResult) {
+	defer wg.Done()
+	r, err := n.Attach(ctx, ns)
+	rc <- asynchAttachResult{res: r, err: err}
+}
+
 func (c *libcni) attachNetworks(ctx context.Context, ns *Namespace) ([]*types100.Result, error) {
+	var wg sync.WaitGroup
+	var lastError error
 	var results []*types100.Result
+	rc := make(chan asynchAttachResult)
+
 	for _, network := range c.Networks() {
-		r, err := network.Attach(ctx, ns)
-		if err != nil {
-			return nil, err
+		wg.Add(1)
+		go asynchAttach(ctx, network, ns, &wg, rc)
+	}
+
+	for range c.Networks() {
+		rs := <-rc
+		if rs.err != nil {
+			lastError = rs.err
```

Comment:
I'm not familiar with what type of errors could occur here. Often times it is best to use the first error for parallel actions rather than last, since the first error could cascade failures to the other jobs, such as cancellation.

Reply:
another good point.. done

Reply:
also ordered the results in c.Networks() index sequence in case someone cares... noting that the unit test did care..